### PR TITLE
Add milvus-sdk-go

### PR DESCRIPTION
**Pull request by fishpenguin**
#845

### CODE REVIEW

File: milvus/ClientProxy.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/testdata"
+	"log"
+)
+
+var (
+	tls                = flag.Bool("tls", false, "Connection uses TLS if true, else plain TCP")
+	caFile             = flag.String("ca_file", "", "The file containing the CA root cert file")
+	serverHostOverride = flag.String("server_host_override", "x.test.youtube.com", "The server name use to verify the hostname returned by TLS handshake")
```
Comment (JinHai-CN) [inline milvus/ClientProxy.go]: why youtube.com?

File: milvus/ClientProxy.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	pb "github.com/milvus-io/milvus-sdk-go/milvus/grpc/gen"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/testdata"
+	"log"
+)
+
+var (
+	tls                = flag.Bool("tls", false, "Connection uses TLS if true, else plain TCP")
+	caFile             = flag.String("ca_file", "", "The file containing the CA root cert file")
```
Comment (JinHai-CN) [inline milvus/ClientProxy.go]: gprc need tls / ca?

File: milvus/ClientProxy.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	serverHostOverride = flag.String("server_host_override", "x.test.youtube.com", "The server name use to verify the hostname returned by TLS handshake")
+)
+
+type Milvusclient struct {
+	MClient MilvusGrpcClient
```
Comment (JinHai-CN) [inline milvus/ClientProxy.go]: struct already has Milvus client concept, why still use MClient, suggest change it to Instance.

File: milvus/ClientProxy.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	}
+	return status{int32(grpcPartitionList.GetStatus().GetErrorCode()), grpcPartitionList.GetStatus().GetReason()}, partitionList
+}
+
+func (client *Milvusclient) DropPartition(partitionParam PartitionParam) Status {
+	grpcPartitionParam := pb.PartitionParam{partitionParam.TableName, partitionParam.PartitionName,
+		partitionParam.PartitionTag, struct{}{}, nil, 0}
+	grpcStatus := client.MClient.DropPartition(grpcPartitionParam)
+	return status{int32(grpcStatus.GetErrorCode()), grpcStatus.Reason}
+}
```
Comment (JinHai-CN) [inline milvus/ClientProxy.go]: Two more interface is provided by Yudong, please double check it.

File: milvus/MilvusClient.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+//TableSchema informations of a table
+type TableSchema struct {
+	// TableName table name
+	TableName string
+	// Dimension Vector dimension, must be a positive value
+	Dimension int64
+	// IndexFileSize Index file size, must be a positive value
+	IndexFileSize int64
+	// MetricType Index metric type
+	MetricType int32
```
Comment (JinHai-CN) [inline milvus/MilvusClient.go]: Why not use int64?

File: milvus/MilvusClient.go [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+}
+
+// IndexParam index parameters
+type IndexParam struct {
+	// TableName table name for create index
+	TableName string
+	// IndexType create index type
+	IndexType IndexType
+	// Nlist index nlist
+	Nlist int32
```
Comment (JinHai-CN) [inline milvus/MilvusClient.go]: Why not use int64

File: milvus/Status.go [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+// Status for SDK interface return
+type Status interface {
+	Ok() bool
+	GetStatus() status
+	GetMessage() string
+}
+
+type status struct {
+	ErrorCode int32
```
Comment (JinHai-CN) [inline milvus/Status.go]: Why not in64

File: milvus/ClientProxy.go (hunk 2 above)
Comment (fishpenguin) [reply milvus/ClientProxy.go]: tls / ca is an optional but not necessary function for grpc. I put it here just as a template, in fact, I didn't use it in our project.

File: milvus/ClientProxy.go (hunk 1 above)
Comment (fishpenguin) [reply milvus/ClientProxy.go]: I put it here just as a template. I will delete it later

File: milvus/MilvusGrpcClient.go [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+	return &milvusGrpcClient{client}
+}
+
+func (grpcClient *milvusGrpcClient) CreateTable(tableSchema pb.TableSchema) pb.Status {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	reply, err := grpcClient.client.CreateTable(ctx, &tableSchema)
+	if err != nil {
+		log.Println("CreateTable rpc failed: " + err.Error())
+	}
```
Comment (ReigenAraka) [inline milvus/MilvusGrpcClient.go]: 这个 error 还是要返回的

File: examples/MilvusClientExample.go [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+var indexFileSize int64 = 1024
+var metricType int64 = int64(milvus.L2)
+var nq int64 = 100
+var nprobe int64 = 64
+var nb int64 = 100000
+var topk int64 = 100
+var nlist int64 = 16384
+
+func example(address string, port string) {
+	var grpcClient milvus.Milvusclient
```
Comment (JinHai-CN) [inline examples/MilvusClientExample.go]: 4 characters tab is OK, why they are 8 characters?

File: milvus/MilvusGrpcClient.go [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	reply, err := grpcClient.client.CreateTable(ctx, &tableSchema)
+	if err != nil {
+		log.Println("CreateTable rpc failed: " + err.Error())
+	}
+	return *reply
+}
+
+func (grpcClient *milvusGrpcClient) HasTable(tableName pb.TableName) pb.BoolReply {
```
Comment (JinHai-CN) [inline milvus/MilvusGrpcClient.go]: So, why the error status isn't return value?

File: examples/MilvusClientExample.go (hunk 9 above)
Comment (fishpenguin) [reply examples/MilvusClientExample.go]: go fmt set=8 defaul. This show 8 characters in github repo.

File: milvus/MilvusGrpcClient.go [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+
+	Cmd(command pb.Command) (pb.StringReply, error)
+
+	DeleteByDate(deleteByDateParam pb.DeleteByDateParam) (pb.Status, error)
+
+	PreloadTable(tableName pb.TableName) (pb.Status, error)
+}
+
+type milvusGrpcClient struct {
+	client pb.MilvusServiceClient
```
Comment (JinHai-CN) [inline milvus/MilvusGrpcClient.go]: change client to instance.

**Pull request merged**